### PR TITLE
FT for CreateTeamCityBuildChain

### PR DESCRIPTION
**Pull request by adellinekania**
Add functional tests for the `CreateTeamCityBuildChain` covering these scenarios:
- Build chains for components with varying distribution types (explicit/implicit, external/internal)
- Build chains for components with different Java versions
- Build chains for components with different build systems (Gradle, Maven, provided, unsupported)

Tests require running the `components-registry-service` in a Docker container, with the `components-registry-service.version` parameter specified.

### CODE REVIEW

**Review by ozonophore — APPROVED** [review]

**Review by ozonophore — DISMISSED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -66,6 +70,7 @@ dependencies {
     implementation("com.github.ajalt.clikt:clikt:4.4.0")
     implementation("org.octopusden.octopus.octopus-external-systems-clients:teamcity-client:${properties["teamcity-client.version"]}")
     implementation("org.octopusden.octopus.infrastructure:components-registry-service-client:${properties["components-registry-service-client.version"]}")
+    implementation("org.octopusden.octopus.infrastructure:component-resolver-core:${properties["component-resolver-core.version"]}")
```
Comment (pgorbachev) [inline build.gradle.kts:73]: what do we need it for? crs client should be enough

File: gradle.properties [hunk 2]
```diff
@@ -1,5 +1,6 @@
 teamcity-client.version=2.0.47
 components-registry-service-client.version=2.0.32
+component-resolver-core.version=2.0.34
```
Comment (pgorbachev) [inline gradle.properties:3]: Do we need it?

**Review by pgorbachev — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (adellinekania) [reply build.gradle.kts:73]: Yes, this dependency is needed for configuring the components registry. We specifically use `ComposedConfigScript` from this dependency within `Aggregator.groovy`, as well as `BuildSystem` and `RepositoryType` for the component configurations. Without it, the components registry service fails to start properly.

File: gradle.properties (hunk 2 above)
Comment (adellinekania) [reply gradle.properties:3]: Yes, this dependency is needed for configuring the components registry. We specifically use `ComposedConfigScript` from this dependency within `Aggregator.groovy`, as well as `BuildSystem` and `RepositoryType` for the component configurations. Without it, the components registry service fails to start properly.

File: gradle.properties [hunk 3]
```diff
@@ -1,6 +1,5 @@
 teamcity-client.version=2.0.47
 components-registry-service-client.version=2.0.32
-component-resolver-core.version=2.0.34
```
Comment (pgorbachev) [inline gradle.properties]: Please update to the latest CRS version

**Review by pgorbachev — CHANGES_REQUESTED** [review]

**Review by pgorbachev — APPROVED** [review]

**Review by CamaradeRoman — APPROVED** [review]

**Pull request merged**